Signal/src/view controllers/PrivacySettingsTableViewController.m

```diff
@@ -75,6 +94,32 @@ - (void)loadView {
     [self.blockOnIdentityChangeSwitch addTarget:self
                                          action:@selector(didToggleBlockOnIdentityChangeSwitch:)
                                forControlEvents:UIControlEventTouchUpInside];
+        
+    self.enableTouchIDSecurityCell.accessoryView = self.enableTouchIDSecuritySwitch;
+    self.enableTouchIDSecurityCell.userInteractionEnabled = YES;
+    
+    //Clear History Log Cell
+    self.clearHistoryLogCell = [[UITableViewCell alloc]init];
+    self.clearHistoryLogCell.textLabel.text = NSLocalizedString(@"SETTINGS_CLEAR_HISTORY", @"");
+    self.clearHistoryLogCell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+    
+    [self detectTouchID];
+    [self validateSecuritySwitches];
+}
+
+/// Enables or disables the TouchID switch based on hardware availability.
+- (void)detectTouchID {
+    if (TouchIDManager.shared.isTouchIDAvailable) {
+        self.enableTouchIDSecuritySwitch.enabled = YES;
+    } else {
+        // Cannot use touchID at this time / on this device
+        self.enableTouchIDSecuritySwitch.enabled = NO;
+    }
+    
+#if DEBUG
```

Comment:
I'd prefer not to - the more we diverge from RELEASE the more likely we are to break shit later and not notice =(